src/main/java/seedu/address/model/ModelMethods.java

```diff
@@ -0,0 +1,76 @@
+package seedu.address.model;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.model.person.Person;
+import seedu.address.model.visit.Visit;
+
+public class ModelMethods {
+
+    /**
+     * Returns a HashSet of ids of persons as given in the argument.
+     */
+    public static HashSet<Index> getIdHashSetFromPersonsList(List<Person> persons) {
+        HashSet<Index> ids = new HashSet<>();
+        for (Person person : persons) {
+            ids.add(person.getId());
+        }
+        return ids;
+    }
+
+    /**
+     * Returns a ArrayList of location ids of visits as given in the argument.
+     */
+    public static ArrayList<Index> getLocationIdsFromInfectedVisitList(List<Visit> visits) {
+        HashMap<Index, Integer> infectedLocations = new HashMap<>();
+        for (Visit visit : visits) {
+            Index id = visit.getLocationId();
+            if (infectedLocations.containsKey(id)) {
+                infectedLocations.put(id, infectedLocations.get(id) + 1);
+            } else {
+                infectedLocations.put(id, 1);
+            }
+        }
+        HashMap<Index, Integer> sortedInfectedLocations = sortByValues(infectedLocations);
+        return new ArrayList<>(sortedInfectedLocations.keySet());
+    }
```

Comment:
Perhaps these methods could be shifted to info handler class? This can help to consolidate all information handling methods to one class.

Reply:
Done. In fact, I think that all methods in ModelMethods class can be moved to InfoHandler class and I have shfited all and refactor the test accordingly.